systemtest/src/test/java/io/strimzi/systemtest/dynamicconfiguration/DynamicConfigurationIsolatedST.java

```diff
@@ -0,0 +1,292 @@
+            basicExternalKafkaClientTls.receiveMessagesTls(Constants.GLOBAL_CLIENTS_EXCEPT_ERROR_TIMEOUT);
+            LOGGER.error("Producer & Consumer did not send and receive messages because external listener is set to plain communication");
+        });
+
+        LOGGER.info("Updating listeners of Kafka cluster");
+        KafkaResource.replaceKafkaResource(CLUSTER_NAME, k -> {
+            KafkaListeners updatedKl = new KafkaListenersBuilder()
+                    .withNewKafkaListenerExternalNodePort()
+                        .withNewKafkaListenerAuthenticationTlsAuth()
+                        .endKafkaListenerAuthenticationTlsAuth()
+                    .endKafkaListenerExternalNodePort()
+                    .build();
+            KafkaClusterSpec kafkaClusterSpec = k.getSpec().getKafka();
+            kafkaClusterSpec.setListeners(updatedKl);
+        });
+
+        kafkaPods = StatefulSetUtils.waitTillSsHasRolled(kafkaStatefulSetName(CLUSTER_NAME), KAFKA_REPLICAS, kafkaPods);
+
+        basicExternalKafkaClientTls.verifyProducedAndConsumedMessages(
+                basicExternalKafkaClientTls.sendMessagesTls(),
+                basicExternalKafkaClientTls.sendMessagesTls()
+        );
+
+        assertThrows(Exception.class, () -> {
+            basicExternalKafkaClientPlain.sendMessagesPlain(Constants.GLOBAL_CLIENTS_EXCEPT_ERROR_TIMEOUT);
+            basicExternalKafkaClientPlain.receiveMessagesPlain(Constants.GLOBAL_CLIENTS_EXCEPT_ERROR_TIMEOUT);
+            LOGGER.error("Producer & Consumer did not send and receive messages because external listener is set to tls communication");
+        });
+
+        LOGGER.info("Updating listeners of Kafka cluster");
+        KafkaResource.replaceKafkaResource(CLUSTER_NAME, k -> {
+            KafkaListeners updatedKl = new KafkaListenersBuilder()
+                    .withNewKafkaListenerExternalNodePort()
+                        .withTls(false)
+                    .endKafkaListenerExternalNodePort()
+                    .build();
+            KafkaClusterSpec kafkaClusterSpec = k.getSpec().getKafka();
+            kafkaClusterSpec.setListeners(updatedKl);
+        });
+
+        StatefulSetUtils.waitTillSsHasRolled(kafkaStatefulSetName(CLUSTER_NAME), KAFKA_REPLICAS, kafkaPods);
+
+        assertThrows(Exception.class, () -> {
+            basicExternalKafkaClientTls.sendMessagesTls(Constants.GLOBAL_CLIENTS_EXCEPT_ERROR_TIMEOUT);
+            basicExternalKafkaClientTls.receiveMessagesTls(Constants.GLOBAL_CLIENTS_EXCEPT_ERROR_TIMEOUT);
+            LOGGER.error("Producer & Consumer did not send and receive messages because external listener is set to plain communication");
+        });
+
+        basicExternalKafkaClientPlain.verifyProducedAndConsumedMessages(
+                basicExternalKafkaClientPlain.sendMessagesPlain(),
+                basicExternalKafkaClientPlain.receiveMessagesPlain()
+        );
+    }
+
+    private void updateAndVerifyDynConf(String dynConfValue) {
```

Comment:
This has generic name, but it is not generic method. It should ba then maybe named `updateAndVerifyDynUncleanLeaderElectionConf`

Reply:
I have refactored that method to be more generic :)